docs/Java-SpringBoot-Developer.md

```diff
@@ -0,0 +1,369 @@
+# Java/Spring Boot Developer Documentation
+
+## Index:
+1. [Managed Identity and Key Vault](#managed-identity-and-key-vault)
+    - [Key Vault](#key-vault)
+2. [Key Rotation](#key-rotation) - WIP
+3. [Cosmos DB](#cosmos-db)
+    - [Spring Repository Pattern](#spring-repository-pattern)
+    - [Partition Key Function](#partition-key-usage-in-spring)
+4. [AKS Pod Identity Support](#aks-pod-identity-support) - WIP
+5. [Versioning](#versioning)
+6. [Application Insights](#application-insights)
+
+
+## Managed Identity and Key Vault
+
+After creating a Managed Identity for the Helium web app and assigning get and list secret permissions to Key Vault, the following code successfully authenticates using Managed Identity to create the Key Vault Client. Leveraging Managed Identity in this way eliminates the need to store any credential information in app code. 
+
+### Key Vault 
+
+If you need access to Key Vault in your app, you can retrieve the Key Vault Client from Spring framework's DI rather than have to track credentials and create a new connection.
+
+#### Adding Key Vault via Spring Configuration
+
+Add the dependency "azure-keyvault-secrets-spring-boot-starter" and "azure-client-authentication" to the maven POM file
+
+[POM.xml] (https://github.com/microsoft/helium-java/blob/master/pom.xml) 
+
```

Comment:
Change to: [POM.xml](https://github.com/microsoft/helium-java/blob/master/pom.xml) so link renders properly (remove space)